src/tox/config.py

```diff
@@ -611,7 +611,16 @@ def passenv(testenv_config, value):
         # Flatten the list to deal with space-separated values.
         value = list(itertools.chain.from_iterable([x.split(" ") for x in value]))
 
-        passenv = {"PATH", "PIP_INDEX_URL", "LANG", "LANGUAGE", "LD_LIBRARY_PATH"}
+        passenv = {
+            "PATH",
+            "PIP_INDEX_URL",
+            "LANG",
+            "LANGUAGE",
+            "LD_LIBRARY_PATH",
+            "TOX_WORK_DIR",
+            "TOX_ENV_NAME",
+            "TOX_ENV_WORK_DIR",
```

Comment:
hmmm, seems odd that they'd get whitelisted into `passenv` as well -- I guess that's an order-of-operations issue probably? (depending on when the variables get set)

if we're going forward with this, can we make tox augment `setenv` and then we don't need to worry about `passenv`?

Reply:
Because I wanted these environment variables to be usable across the tool lifecycle, not just the commands section, I change the values in the ``os.environ``. As such in order for the commands to inherit it we need them to be in the whitelist.